### PR TITLE
RN for v5.4.0

### PR DESCRIPTION
**Pull request by Serdaro**

### CODE REVIEW

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 1]
```diff
@@ -0,0 +1,352 @@
+= 5.4.0 Release Notes
+
+CAUTION: Starting with this release of Hazelcast Platform, the minimum supported Java version is 17.
+
+== New Features
+
+* [.enterprise]*Enterprise* **User Code Namespaces**: Enable deployment, undeployment and redeployment of your custom classes. See the xref:clusters:user-code-namespaces.adoc[User Code Namespaces documentation].
```
Comment (rebekah-lawrence) [inline docs/modules/release-notes/pages/5-4-0.adoc:7]:
```suggestion
* [.enterprise]*Enterprise* **User Code Namespaces**: Enable deployment and redeployment of your custom classes. See the xref:clusters:user-code-namespaces.adoc[User Code Namespaces documentation].
```

**Review by rebekah-lawrence — COMMENTED** [review]
Just a small change for UCN

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 2]
```diff
@@ -0,0 +1,352 @@
+= 5.4.0 Release Notes
+
+CAUTION: Starting with this release of Hazelcast Platform, the minimum supported Java version is 17.
+
+== New Features
+
+* [.enterprise]*Enterprise* **User Code Namespaces**: Enable deployment and redeployment of your custom classes. See the xref:clusters:user-code-namespaces.adoc[User Code Namespaces documentation].
```
Comment (oliverhowell) [inline docs/modules/release-notes/pages/5-4-0.adoc:7]:
```suggestion
* [.enterprise]*Enterprise* **User Code Namespaces**: Enable deployment and redeployment of your custom classes. See the xref:clusters:user-code-namespaces.adoc[User Code Namespaces documentation].
* [.enterprise]*Enterprise* **Thread-Per-Core (TPC)**: TPC now generally available. You can enable this on clients and cluster members for improved performance. See the xref:cluster-performance:thread-per-core-tpc.adoc[Thread-Per-Core (TPC) documentation].
```

**Review by oliverhowell — APPROVED** [review]
Adding suggestion for TPC going GA - can you review and add @Serdaro ?

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+* Changed the return values for `getMethodName()` of the following message task classes:
+** `MapFetchKeysMessageTask`: `iterator` -> `SecurityInterceptorConstants.ITERATOR_FETCH_KEYS`
+** `MapFetchWithQueryMessageTask.`: `iterator` -> `SecurityInterceptorConstants.ITERATOR_FETCH_WITH_QUERY`
+** `MapProjectionWithPredicateMessageTask`: `projectWithPredicate` -> `SecurityInterceptorConstants.PROJECT`
+** `ScheduledExecutorGetAllScheduledMessageTask`: `getAllScheduled` -> `SecurityInterceptorConstants.GET_ALL_SCHEDULED_FUTURES`
+** `ScheduledExecutorSubmitToPartitionMessageTask`: `submitToPartition` -> `SecurityInterceptorConstants.SCHEDULE_ON_PARTITION`
+** `ScheduledExecutorSubmitToTargetMessageTask`: `submitToAddress` -> `SecurityInterceptorConstants.SCHEDULE_ON_MEMBER`
+** `ScheduledExecutorTaskGetResultFromPartitionMessageTask`: `getResultTimeout` -> `SecurityInterceptorConstants.GET_RESULT`
+** `ScheduledExecutorTaskGetResultFromTargetMessageTask`: `getResultTimeout` -> `SecurityInterceptorConstants.GET_RESULT`
+** `SetGetAllMessageTask`: `iterator` -> `SecurityInterceptorConstants.GET_ALL`
```
Comment (vbekiaris) [inline docs/modules/release-notes/pages/5-4-0.adoc:29]: This change was subsequently reverted in https://github.com/hazelcast/hazelcast/pull/25551. Can you please check and amend the list?

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+** `TransactionalListAddMessageTask`: `set` -> `SecurityInterceptorConstants.ADD`
+** `TransactionalMultiMapRemoveEntryMessageTask`: `removeEntry` -> `SecurityInterceptorConstants.REMOVE`
++
+https://github.com/hazelcast/hazelcast/pull/25020[#25020]
+* Renamed the service port for Hazelcast clusters deployed in Kubernetes environments to `hazelcast`.
+The previous name, `hazelcast-service-port`, caused member auto-discovery for embedded deployments to fail.
+https://github.com/hazelcast/hazelcast/pull/24834[#24834], https://github.com/hazelcast/hazelcast/issues/24705[#24705], https://github.com/hazelcast/hazelcast/issues/24688[#24688]
+* Fixed an issue where Hazelcast was not fast-failing when a member is started with a blank public address. This has been fixed by introducing a configuration validation
```
Comment (vbekiaris) [inline docs/modules/release-notes/pages/5-4-0.adoc:37]:
```suggestion
* Fixed an issue where Hazelcast was not failing fast when a member is started with a blank public address. This has been fixed by introducing a configuration validation
```
?

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+https://github.com/hazelcast/hazelcast/pull/25348[#25348]
+* Added support for permission subtraction (deny permissions) in client connections. See xref:security:native-client-security.adoc#deny-permissions[Deny Permissions].
+https://github.com/hazelcast/hazelcast/pull/25154[#25154]
+* Added the boolean `forceCertValidation` property to the security configuration to initiate a remote certificate validity check. #6235
+
+=== Storage
+
+* Improved the hit/miss cache statistics counter performance.
+https://github.com/hazelcast/hazelcast/pull/25146[#25146]
+* [.enterprise]*Enterprise* Tiered Store can now be used for map loaders.
```
Comment (vbekiaris) [inline docs/modules/release-notes/pages/5-4-0.adoc:166]:
```suggestion
* [.enterprise]*Enterprise* Tiered Store can now be used with map loaders.
```

Previously we wouldn't allow a map configured with a map store to be also configured with tiered storage. Now users can enable both tiered storage AND a map store for an IMap.

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+* Check for existing resources for Mongo connector is now done only once; previously, it was performed on every processor.
+https://github.com/hazelcast/hazelcast/pull/24953[#24953]
+* Hazelcast JDBC connector now supports Microsoft SQL server.
+https://github.com/hazelcast/hazelcast/pull/25071[#25071]
+* Added the ability to configure the pool size of a MongoDB data connection. See xref:sql:mapping-to-mongo.adoc#creating-a-mongodb-mapping[Creating a MongoDB Mapping].
+https://github.com/hazelcast/hazelcast/pull/25027[#25027]
+
+=== Data Structures
+
+* Added support of specifying the maximum size of a CPMap.
```
Comment (vbekiaris) [inline docs/modules/release-notes/pages/5-4-0.adoc:118]: I think this and the following item can be skipped - they are just parts of introducing the CPMap data structure.

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+* Introduced a cluster state check to improve the removal of expired map/cache entries from the cluster.
+The removal operation is no longer executed if the cluster is in passive state.
+https://github.com/hazelcast/hazelcast/pull/24808[#24808]
+* Added the `IMap.localValues()` and `IMap.localValues(Predicate)` methods for a faster access to locally owned values in maps.
+https://github.com/hazelcast/hazelcast/pull/24673[#24763]
+* Added the `deleteAsync()` for maps to asynchronously remove a provided map entry key.
+https://github.com/hazelcast/hazelcast/pull/23509[#23509]
+* Fixed the Javadoc for caches where it was referring incorrectly to statistics instead of management, for the `setManagementEnabled()` method.
+https://github.com/hazelcast/hazelcast/pull/22575[#22575]
+* Implemented RAFT lifecycle interfaces for CPMap. #6800
```
Comment (vbekiaris) [inline docs/modules/release-notes/pages/5-4-0.adoc:133]: This item is also just normal part of implementing a CP data structure, so nothing notable for users to see, I 'd rather remove it.

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+* Added the `getCPObjectInfos()` method to create a snapshot of all existing CP objects for a given service on the group leader. #615
+* Added the `getCPGroupIds()` method to the CP Subsystem API to allow listing all data structures within a CP group. #591
+* The "last update time" record of the map entries recovered from the disk persistence is not used anymore when calculating the entries' expiration times. #233
+* [.enterprise]*Enterprise* Added the ability to check whether the `ADVANCED_CP` feature is present in the Hazelcast Platform Enterprise license, to enable CPMap. #157
```
Comment (vbekiaris) [inline docs/modules/release-notes/pages/5-4-0.adoc:137]: just a license check PR, not RN-worthy I think. The requirements for using CPMap are anyway listed in the CPMap documentation page.

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+
+=== Networking
+
+* Added the ability to evaluate priorities while picking member addresses; when interfaces are not configured, the first possible IP address is no longer used. Instead, all addresses are evaluated and the one with the highest priority (IPv6 address) is selected.
+https://github.com/hazelcast/hazelcast/pull/25305[#25305]
+* Added the `demoteLocalDataMember()` method to convert members holding data to lite members, enabling a cluster to be alive while there is no data in it.
+https://github.com/hazelcast/hazelcast/pull/24617[#24617]
+* Improved the duration of joins by the clients to the cluster; the clients can now join instantaneously without waiting by internally delaying the migrations asynchronously.
+https://github.com/hazelcast/hazelcast/pull/17582[#17582]
+* Added support for configuration of `CLASS` type resources for user code namespaces. #702
```
Comment (vbekiaris) [inline docs/modules/release-notes/pages/5-4-0.adoc:179]: Not RN-worth, just part of UCN implementation

**Review by vbekiaris — CHANGES_REQUESTED** [review]

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+** `MapFetchWithQueryMessageTask.`: `iterator` -> `SecurityInterceptorConstants.ITERATOR_FETCH_WITH_QUERY`
+** `MapProjectionWithPredicateMessageTask`: `projectWithPredicate` -> `SecurityInterceptorConstants.PROJECT`
+** `ScheduledExecutorGetAllScheduledMessageTask`: `getAllScheduled` -> `SecurityInterceptorConstants.GET_ALL_SCHEDULED_FUTURES`
+** `ScheduledExecutorSubmitToPartitionMessageTask`: `submitToPartition` -> `SecurityInterceptorConstants.SCHEDULE_ON_PARTITION`
+** `ScheduledExecutorSubmitToTargetMessageTask`: `submitToAddress` -> `SecurityInterceptorConstants.SCHEDULE_ON_MEMBER`
+** `ScheduledExecutorTaskGetResultFromPartitionMessageTask`: `getResultTimeout` -> `SecurityInterceptorConstants.GET_RESULT`
+** `ScheduledExecutorTaskGetResultFromTargetMessageTask`: `getResultTimeout` -> `SecurityInterceptorConstants.GET_RESULT`
+** `SetGetAllMessageTask`: `iterator` -> `SecurityInterceptorConstants.GET_ALL`
+** `TransactionalListAddMessageTask`: `set` -> `SecurityInterceptorConstants.ADD`
+** `TransactionalMultiMapRemoveEntryMessageTask`: `removeEntry` -> `SecurityInterceptorConstants.REMOVE`
```
Comment (kwart) [inline docs/modules/release-notes/pages/5-4-0.adoc:31]:
```suggestion
* Method names used as parameters in https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/security/SecurityInterceptor.html[`SecurityInterceptor`] checks were reviewed and unified into a single place  - class `com.hazelcast.security.SecurityInterceptorConstants`. Some client messages have the method name changed to reflect their purpose better. Some client messages are newly covered by `SecurityInterceptor` checks now.
```

Here is the list of message tasks that have the value changed in `5.4.0`:
```
TransactionalListAddMessageTask.java
ReplicatedMapAddNearCacheListenerMessageTask.java
MapDestroyCacheMessageTask.java
MapAddNearCacheInvalidationListenerMessageTask.java
AbstractMapPartitionMessageTask.java
MapFetchWithQueryMessageTask.java
MapFetchEntriesMessageTask.java
MapAddListenerMessageTask.java
MapProjectionWithPredicateMessageTask.java
ScheduledExecutorSubmitToPartitionMessageTask.java
ScheduledExecutorTaskGetResultFromTargetMessageTask.java
ScheduledExecutorSubmitToTargetMessageTask.java
ScheduledExecutorGetAllScheduledMessageTask.java
ScheduledExecutorTaskGetResultFromPartitionMessageTask.java
ListGetAllMessageTask.java
XATransactionCommitMessageTask.java
XATransactionPrepareMessageTask.java
XATransactionRollbackMessageTask.java
TransactionCommitMessageTask.java
TransactionCreateMessageTask.java
XATransactionCreateMessageTask.java
XAClearRemoteTransactionMessageTask.java
TransactionRollbackMessageTask.java
XACollectTransactionsMessageTask.java
XAFinalizeTransactionMessageTask.java
PNCounterGetConfiguredReplicaCountMessageTask.java
CacheSizeMessageTask.java
CacheListenerRegistrationMessageTask.java
TransactionalMapSizeMessageTask.java
CacheListenerRegistrationMessageTask.java
CacheDestroyMessageTask.java
CacheGetAndReplaceMessageTask.java
CacheManagementConfigMessageTask.java
CacheAddPartitionLostListenerMessageTask.java
CacheAddEntryListenerMessageTask.java
CacheRemoveInvalidationListenerMessageTask.java
CacheRemovePartitionLostListenerMessageTask.java
CacheCreateConfigMessageTask.java
CacheLoadAllMessageTask.java
CacheGetConfigMessageTask.java
DestroyProxyMessageTask.java
ExecutorServiceSubmitToPartitionMessageTask.java
ExecutorServiceSubmitToAddressMessageTask.java
DurableExecutorRetrieveAndDisposeResultMessageTask.java
DurableExecutorSubmitToPartitionMessageTask.java
DurableExecutorDisposeResultMessageTask.java
DurableExecutorRetrieveResultMessageTask.java
CreateProxyMessageTask.java
TransactionalMultiMapRemoveEntryMessageTask.java
AbstractMultiTargetMessageTask.java
CreateProxiesMessageTask.java
```

**Review by vbekiaris — APPROVED** [review]

**Pull request merged**